pkg/system/xattrs_linux.go

```diff
@@ -22,6 +26,9 @@ func Lgetxattr(path string, attr string) ([]byte, error) {
 		return nil, nil
 	}
 	if errno == unix.ERANGE {
+		if sz > math.MaxUint16 {
+			return nil, unix.E2BIG
+		}
 		dest = make([]byte, sz)
```

Comment:
The issue is that `sz` can still be `-1` here so the above check does not fix.

Reply:
https://github.com/containers/storage/pull/483 looks good. Thanks!